src/platforms/dotnet/guides/google-cloud-functions/index.mdx

```diff
@@ -0,0 +1,190 @@
+- RequestId as tag
+- URL as tag
+- Environment is automatically set (IHostingEnvironment)
+- Release automatically set (`AssemblyInformationalVersionAttribute`, `AssemblyVersion` or environment variable)
+- Request payload can be captured if opt-in
+- Support for EventProcessors registered with DI
+- Support for ExceptionProcessors registered with DI
+- Supports configuration system (bind to `SentryAspNetCoreOptions`)
+- Server OS info sent
+- Server Runtime info sent
+- Request headers sent
+- HTTP Proxy configuration
+- Request body compressed
+- Event flooding protection (429 retry-after and internal bound queue)
+- Assembly Strong named
+
+## Installation
+
+Add the Sentry dependency:
+
+```powershell {tabTitle:Package Manager}
+Install-Package Sentry.Google.Cloud.Functions -Version {{ packages.version('sentry.dotnet.google-cloud-function') }}
+```
+
+```shell {tabTitle:.NET Core CLI}
+dotnet add package Sentry.Google.Cloud.Functions -v {{ packages.version('sentry.dotnet.google-cloud-function') }}
+```
+
+This package extends `Sentry` main SDK. That means that besides the logging-related features, through this package you'll also get access to all API and features available in the main `Sentry` SDK.
+
+<Alert level="warning" title="Note">
+
+Messages logged from assemblies with the name starting with `Sentry` will not generate events.
+
+</Alert>
+
+## Configuration
+
+The simplest way to configure the `Sentry.Google.Cloud.Functions` package is by using the assembly method `FunctionsStartup` to include the type `SentryStartup`, for example:
+```csharp
+//your function.cs
+
+using System.Threading.Tasks;
+using Google.Cloud.Functions.Framework;
+using Google.Cloud.Functions.Hosting;
+using Microsoft.AspNetCore.Http;
+
+[assembly: FunctionsStartup(typeof(SentryStartup))]
+
+public class Function : IHttpFunction
+{
+}
+```
+
+When using `[assembly: FunctionsStartup(typeof(SentryStartup))]`, the framework automatically loads `appsettings.json` and environment variables, binding to `SentryAspNetCoreOptions`.
+
+For example:
+
+```json {filename:appsettings.json}
+  "Sentry": {
+    "Dsn": "___PUBLIC_DSN___",
+    "MaxRequestBodySize": "Always",
+    "SendDefaultPii": true,
+    "MinimumBreadcrumbLevel": "Debug",
+    "MinimumEventLevel": "Warning",
+    "AttachStackTrace": true,
+    "Debug": true,
+    "DiagnosticsLevel": "Error"
+  }
+```
+
+> An example of some of the options that can be configured via `appsettings.json`.
+
+ASP.NET Core will automatically read this environment variable and bind it to the SDK configuration object.
+
+Some of the settings require actual code. For those, like the `BeforeSend` callback you will need to create a new class implementing the abstract class `FunctionsStartup` to set it up:
+
+```csharp
+using Google.Cloud.Functions.Hosting;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Sentry.AspNetCore;
+
+[assembly: FunctionsStartup(typeof(CustomOptions))]
```

Comment:
Is there a chance this is loaded after the SDK initializes?
The reference to the options object is the same so in the case of BeforeSend it might not matter (only if events are captured before this runs) but what about if you set a value that's read during `init` and then not anymore? Like HTTP proxy?

Reply:
`Configure<SentryAspNetCoreOptions>` is only invoked once the SDK is initialized.
If you set a parameter Like an HTTP Proxy, it'll stay set.

Reply:
True so we'll have all these startup registrations get picked up first, all `ConfigureServices` will run and only once `Configure` runs we'll get the configuration callback invoked.
Should work (at least for the second init, since the first happens while setting up the logger only)